tests/datatypes/test_str.py

```diff
@@ -30,13 +30,27 @@ def test_isspace(self):
                 print(s.isspace())
             """)
 
+    def test_isalnum(self):
+        self.assertCodeExecution("""
+            for word in ["", "12", "abc", "abc12", "\u00c4", "\x41", "a@g", "äÆ",
+            "12.2", "'hi'", "hello!!", "gEt_rIgHt_jR."]:
```

Comment:
Please use strings that specifically exercise various edge cases :) I'm not sure what the last one is testing for, compared to the rest.